File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] num, int t) {
        int l=0, r= num.length-1;
        while (l < r) {
            int sum = num[l] + num[r];
            if (sum > t) {
                r--;
            }
            if (sum < t) {
                l++;
            }
            if (sum == t && num[l] != num[r]) {
                return new int[] {l+1, r+1};
            }
        }
        return new int[0];
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const map = new Map()
        for(let i = 0 ; i < nums.length ;i++){
            const remainder = target - nums[i]
            if(map.has(remainder)){
                return [map.get(remainder) , i]
            }
            map.set(nums[i] , i)
        }
        return []
    }
}
